### ISSUE TITLE
Modification of the agents and evaluation

### ISSUE THREAD

**Issue by TimotheeMathieu** [rlberry-py/rlberry#85]
I propose that instead of always having to define an `eval` function in the agents, we have an `eval` parameter that can be either "cumulative_reward", "best_arm_identification",... or "custom" and then if `eval` is custom, the eval function must be specified. 

This would simplify the creation of a basic agent because in the end, we often use the same evaluations. 

Example of resulting template agent :
```python
class MyAgent(Agent):
    name = "MyAgent"

    def __init__(self,
                 env,
                 eval="cumulative_reward",
                 param_1,
                 param_2,
                 param_n,
                 **kwargs):
        Agent.__init__(self, env, eval, **kwargs)

    def fit(self, budget: int):
        """
        ** Must be implemented. **

        Trains the agent, given a computational budget (e.g. number of steps or episodes).
        """
        # code to train the agent
        # ...
        pass

    @classmethod
    def sample_parameters(cls, trial):
        """
        ** Optional **

        Sample hyperparameters for hyperparam optimization using
        Optuna (https://optuna.org/).

        Parameters
        ----------
        trial: optuna.trial
        """
        # Note: param_1 and param_2 are in the constructor.

        # for example, param_1 could be the batch_size...
        param_1 = trial.suggest_categorical('param_1',
                                            [1, 4, 8, 16, 32, 64])
        # ... and param_2 could be a learning_rate
        param_2 = trial.suggest_loguniform('param_2', 1e-5, 1)
        return {
            'param_1': param_1,
            'param_2': param_2,
        }
```

**Comment by TimotheeMathieu** [comment]
You are right, I agree that it must stay fully customizable but my idea was more along the lines of providing some often-used evaluations and for instance this can be cumulative reward for some specific settings.

**Comment by TimotheeMathieu** [comment]
Solved by PR #89
